scala/tempo/src/main/scala/com/databrickslabs/tempo/ml/stat/EMA.scala

```diff
@@ -0,0 +1,167 @@
+package com.databrickslabs.tempo.ml.stat
+
+import com.databrickslabs.tempo.ml.Orderings.Ordering
+import com.databrickslabs.tempo.ml.WindowAlignments.WindowAlignment
+import com.databrickslabs.tempo.ml._
+import com.databrickslabs.tempo.{TSDF, TSStructType}
+import org.apache.spark.ml.param.{DoubleParam, IntParam, LongParam, Param, ParamMap}
+import org.apache.spark.ml.util.Identifiable
+import org.apache.spark.sql.types.{DataTypes, StructField}
+import org.apache.spark.sql.{functions => fn}
+
+import scala.math.pow
+
+class EMA(uid: String)
+	extends TSDFTransformer(uid) with HasMeasureCol with HasOutputCol
+{
+	def this() = this(Identifiable.randomUID("EMA"))
+
+	override def copy(extra: ParamMap): TSDFTransformer = defaultCopy(extra)
+
+	// parameters
+
+	/** @group param */
+	final val N = new IntParam(this,
```

Comment:
This may have been the original name but let's update this to windowLength or something indicative of the logical meaning.

Reply:
done